src/Microsoft.Health.Fhir.Api/Features/Throttling/ThrottlingMiddleware.cs

```diff
@@ -0,0 +1,69 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Configs;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+
+namespace Microsoft.Health.Fhir.Api.Features.Throttling
+{
+    public class ThrottlingMiddleware
+    {
+        private readonly ILogger<ThrottlingMiddleware> _logger;
+        private readonly ThrottlingConfiguration _configuration;
+        private int _requestsInFlight = 0;
+        private RequestDelegate _next;
+
+        public ThrottlingMiddleware(RequestDelegate next, IOptions<ThrottlingConfiguration> configuration, ILogger<ThrottlingMiddleware> logger)
+        {
+            _next = EnsureArg.IsNotNull(next, nameof(next));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            _configuration = EnsureArg.IsNotNull(configuration?.Value, nameof(configuration));
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (_configuration.ExcludedEndpoints.Contains($"{context.Request.Method}:{context.Request.Path.Value?.TrimEnd('/')}", StringComparer.OrdinalIgnoreCase))
```

Comment:
This implementation is falling back to ExcludedEndpoints as an iEnumberable, because you are passing in the comparer. The comparer should be set in the constructor instead.

Reply:
You could also likely achieve a more efficient implementation using a ValueTuple instead of allocating a string.